tests/lib/quiet.sh

```diff
@@ -0,0 +1,18 @@
+# use "quiet foo" when you expect "foo" to produce a lot of output
+# that isn't useful unless foo itself fails.
+quiet() (
+    # note this is a subshell (parens instead of braces around the function)
+    # so this set only affects this function itself
+    { set +x; } >&/dev/null
+
+    tf="$(tempfile)"
+    if "$@" >& "$tf"; then
+        rm -f -- "$tf"
+        return 0
+    else
+        echo "Q: $*" >&2
```

Comment:
"quiet $*" would make it more clear. Suspect "Q:" won't mean much to most readers.